### PR TITLE
Allow referees to submit safeguarding concerns about the candidate

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

We want to allow the referees to say whether the applicant is a suitable candidate to work with children so that a provider can properly assess their application. This the third slice of reworking the referee flow, the previous PR added the `Confirm relationship` page  #1598. 

## Changes proposed in this pull request
This PR adds:
- A `ReferenceSafeguardingForm` to update safeguarding concerns
- The safeguarding page to allow the referees submitting a reason why the candidate should not work with children

### The new page
image

image

## Guidance to review

Reviewing commit-by-commit can be helpful

## Link to Trello card
https://trello.com/c/QlKuThen/1130-build-referee-view-of-candidate-working-with-children

## Things to check
- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/referee_interface/reference_safeguarding_form.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+module RefereeInterface
+  class ReferenceSafeguardingForm
+    include ActiveModel::Model
+    attr_accessor :any_safeguarding_concerns, :safeguarding_concerns, :candidate
+
+    validate :presentce_of_any_safeguarding_concerns
+    validate :presentce_of_safeguarding_concerns
```
Comment (tvararu) [inline app/models/referee_interface/reference_safeguarding_form.rb]:
```suggestion
    validate :presence_of_any_safeguarding_concerns
    validate :presence_of_safeguarding_concerns
```

File: app/models/referee_interface/reference_safeguarding_form.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+module RefereeInterface
+  class ReferenceSafeguardingForm
+    include ActiveModel::Model
+    attr_accessor :any_safeguarding_concerns, :safeguarding_concerns, :candidate
+
+    validate :presentce_of_any_safeguarding_concerns
```
Comment (tvararu) [inline app/models/referee_interface/reference_safeguarding_form.rb]: Isn't this just:

```ruby
validate :any_safeguarding_concerns, presence: true
```

File: config/routes.rb [hunk 3]
```diff
@@ -291,6 +291,9 @@
     get '/' => 'reference#relationship', as: :reference_relationship
     patch '/confirm-relationship' => 'reference#confirm_relationship', as: :confirm_relationship
 
+    get '/safeguarding' => 'reference#safeguarding', as: :safeguarding
+    patch 'confirm-safeguarding' => 'reference#confirm_safeguarding', as: :confirm_safeguarding
```
Comment (tvararu) [inline config/routes.rb]:
```suggestion
    patch '/confirm-safeguarding' => 'reference#confirm_safeguarding', as: :confirm_safeguarding
```

File: app/models/referee_interface/reference_safeguarding_form.rb (hunk 1 above)
Comment (C-gyorfi) [reply app/models/referee_interface/reference_safeguarding_form.rb]: I think I had the same typo on the previous PR  :shame_emoji

File: app/models/referee_interface/reference_safeguarding_form.rb (hunk 2 above)
Comment (C-gyorfi) [reply app/models/referee_interface/reference_safeguarding_form.rb]: There may be an easier way to do this, but I need a custom error message with the candidate name. 

I also tried:
`validate_presence_of :any_safeguarding_concerns, message: "Some error message"`

But it didn't like string interpolation

**Pull request merged**